wweval/R/score.R

```diff
@@ -186,6 +186,109 @@ make_baseline_score_table <- function(all_ww_scores,
+#' threshold for the proportion of forecast dates that a model must have
+#' submitted forecasts to be included in analysis
+#' @param prop_locs_incl_hub Numeric less than 1 indicating the inclusion
+#' threshold for the proportion of the locations we expect that a model
+#' must have subbmited forecasts for to be included in analysis
+#' @param forecast_dates vector of dates formatted in ISO8601 convention
+#' (YYYY-MM-DD) indicating the forecast dates for the analysis
+#' @param locations vector of state abbreviations that we want to ensure the
+#' submitting teams have produced forecasts for
+#' @param project_name name of the Zoltar project, default is
+#' `"COVID-19 Forecasts"`
+#'
+#' @return a vector of character strings indicating the unique model names
+#' that fit the inclusion criteria
+#' @export
+query_and_select_models <- function(prop_dates_for_incl_hub,
+                                    prop_locs_for_incl_hub,
+                                    forecast_dates,
+                                    locations,
+                                    project_name = "COVID-19 Forecasts") {
+  # get state abbreviation codes
+  state_codes <- cfaforecastrenewalww::loc_abbr_to_flusight_code(
+    unique(locations)
+  )
+
+  if (prop_dates_for_incl_hub > 1) {
+    cli::cli_abort(c(
+      "Proportion of forecast dates required for hub inclusion",
+      "must be less than 1."
+    ))
+  }
+
+  if (prop_locs_for_incl_hub > 1) {
+    cli::cli_abort(c(
+      "Proportion of locations required for hub inclusion",
+      "must be less than 1."
+    ))
+  }
+
+  zoltar_connection <- zoltr::new_connection()
+  zoltr::zoltar_authenticate(
+    zoltar_connection, get_secret("Z_USERNAME"),
+    get_secret("Z_PASSWORD")
+  )
+
+  # list of project on zoltar
+  the_projects <- zoltr::projects(zoltar_connection)
+
+  # Grabbing a specific project
+  project_url <- the_projects[the_projects$name == !!project_name, "url"]
+  the_project_info <- zoltr::project_info(zoltar_connection, project_url)
+
+  # get the models
+  the_models <- zoltr::models(zoltar_connection, project_url)
+
+  # Submit query, poll job, get job data
+
+  forecast_data <- zoltr::do_zoltar_query(
+    zoltar_connection = zoltar_connection,
+    project_url = project_url,
+    query_type = "forecasts",
+    models = NULL, # all models by default
+    units = state_codes,
+    # We could query all of them, but this was very slow. This ensures
+    # that the forecasts submitted have at leastr eached 28 days.
+    targets = c("28 day ahead inc hosp"),
+    types = "quantile",
+    timezeros = forecast_dates
+  )
+
+  n_unique_forecasts <- forecast_data |>
+    dplyr::distinct(timezero) |>
+    dplyr::pull() |>
+    length()
+
+  n_forecasts_per_model <- forecast_data |>
+    dplyr::distinct(timezero, model) |>
+    dplyr::group_by(model) |>
+    dplyr::summarize(
+      n_forecast_dates = dplyr::n(),
+      prop_present = n_forecast_dates / n_unique_forecasts
```

Comment:
```suggestion
      prop_present = n_forecast_dates / !!n_unique_forecasts
```

Reply:
Proportion of locations not implemented?

Reply:
My bad I thought I moved it over from the scratch file